internal/providers/pluginfw/products/permissions/object_mutex_test.go

```diff
@@ -0,0 +1,167 @@
+package permissions
+
+import (
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestObjectMutexManager_LockUnlock(t *testing.T) {
+	manager := &objectMutexManager{
+		mutexes: make(map[string]*sync.Mutex),
+	}
+
+	objectID := "/clusters/test-cluster"
+
+	// First lock should succeed immediately
+	manager.Lock(objectID)
+
+	// Verify mutex was created
+	manager.mapLock.Lock()
```

Comment:
You don't need to lock this here; in the context of your test, this is only accessed by one go-routine.